libcxx/test/std/containers/sequences/vector/exception_construct_at_end.pass.cpp

```diff
@@ -0,0 +1,350 @@
+    }
+    check_new_delete_called();
+  }
+#endif
+
+  // void resize(size_type __sz)
+  {
+    // cap < size
+    throw_context ctx;
+    std::vector<throw_element> v;
+    v.reserve(5);
+    try {
+      v.resize(4);
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+  // void resize(size_type __sz, const_reference __x)
+  {
+    // cap < size
+    throw_context ctx(3);
+    std::vector<throw_element> v;
+    v.reserve(5);
+    try {
+      throw_element e;
+      v.resize(4, e);
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+  // void assign(_ForwardIterator __first, _ForwardIterator __last)
+  {
+    // new size <= cap && new size > size
+    throw_context ctx(4);
+    std::vector<throw_element> v;
+    v.reserve(3);
+    try {
+      std::vector<throw_element> data(2);
+      v.assign(data.begin(), data.end());
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+  {
+    // new size > cap
+    throw_context ctx(4);
+    std::vector<throw_element> v;
+    try {
+      std::vector<throw_element> data(2);
+      v.assign(data.begin(), data.end());
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+#if TEST_STD_VER >= 23
+  // void assign_range(_Range&& __range)
+  {
+    throw_context ctx(5);
+    std::vector<throw_element> v;
+    try {
+      std::vector<throw_element> r(3);
+      v.assign_range(r);
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+#endif
+
+#if TEST_STD_VER > 3
+  // vector& operator=(initializer_list<value_type> __il)
+  {
+    throw_context ctx(5);
+    std::vector<throw_element> v;
+    try {
+      throw_element e;
+      v = {e, e};
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+#endif
+
+  // vector<_Tp, _Allocator>& vector<_Tp, _Allocator>::operator=(const vector& __x)
+  {
+    throw_context ctx(4);
+    std::vector<throw_element> v;
+    try {
+      std::vector<throw_element> data(2);
+      v = data;
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+  // iterator insert(const_iterator __position, _ForwardIterator __first, _ForwardIterator __last)
+  {
+    throw_context ctx(6);
+    std::vector<throw_element> v;
+    v.reserve(10);
+    try {
+      std::vector<throw_element> data(3);
+      v.insert(v.begin(), data.begin(), data.end());
+    } catch (int) {
+    }
+    CHECK_NEW_DELETE_DIFF(1);
+  }
+  check_new_delete_called();
+
+#if TEST_STD_VER >= 23
+  // iterator insert_range(const_iterator __position, _Range&& __range)
+  {
+    throw_context ctx(3);
+    std::vector<throw_element> v;
+    try {
+      std::vector<throw_element> data(2);
+      v.insert_range(v.begin(), data);
+    } catch (int) {
+    }
+    check_new_delete_called();
+  }
+#endif
+
+  return 0;
+}
```

Comment:
We conventionally make every source/header file end in empty line, unless it's intentionally to test lack of ending empty line.
```suggestion
}

```